caas/kubernetes/provider/k8s.go

```diff
@@ -1048,39 +1056,24 @@ func (k *kubernetesClient) EnsureService(
 		}
 		cleanups = append(cleanups, func() { k.deleteSecret(imageSecretName, "") })
 	}
-
 	// Add a deployment controller or stateful set configured to create the specified number of units/pods.
 	// Defensively check to see if a stateful set is already used.
-	var useStatefulSet bool
-	if params.Deployment.DeploymentType != "" {
-		useStatefulSet = params.Deployment.DeploymentType == caas.DeploymentStateful
-	} else {
-		useStatefulSet = len(params.Filesystems) > 0
-	}
-	statefulsets := k.client().AppsV1().StatefulSets(k.namespace)
-	existingStatefulSet, err := statefulsets.Get(deploymentName, v1.GetOptions{IncludeUninitialized: true})
-	if err != nil && !k8serrors.IsNotFound(err) {
-		return errors.Trace(err)
-	}
-	if !useStatefulSet {
-		useStatefulSet = err == nil
-		if useStatefulSet {
-			logger.Debugf("no updated filesystems but already using stateful set for %v", appName)
+	if params.Deployment.DeploymentType == "" {
+		// TODO(caas): we should really change `params.Deployment` to be required.
```

Comment:
The current behaviour is to default to stateless if there's no storage definitions and stateful otherwise. I'm not sure that's a bad thing. Most people will have no need to worry about setting deployment type.